### ISSUE TITLE
[apex] AvoidFutureAnnotation rule missing in PMD 7.20.0

### ISSUE THREAD

**Issue by namrata111f** [pmd/pmd#6405]
**Affects PMD Version:** 7.20.0

**Rule:** [AvoidFutureAnnotation](https://docs.pmd-code.org/latest/pmd_rules_apex_bestpractices.html#avoidfutureannotation)

**Description:**

The rule `AvoidFutureAnnotation` exists and works correctly in PMD 7.19.0, but produces zero violations in PMD 7.20.0 on the same codebase. The rule appears to have been silently removed or broken without any deprecation warning or documentation in the release notes. When listing available rules in PMD 7.20.0, this rule does not appear.

**Code Sample demonstrating the issue:**

​```apex
public class AtFutureRecipes {
    
    @future
    public static void atFutureMethodWithoutCalloutPrivileges() {
        System.debug('This method runs asynchronously');
    }

    @future(callout=true)
    public static void atFutureMethodWithCalloutPrivileges(String url) {
        Http h = new Http();
        HttpRequest request = new HttpRequest();
        request.setEndpoint(url);
        HttpResponse response = h.send(request);
    }
}
​```

**Expected outcome:**

PMD should report a violation at line 3 and line 8 (the `@future` annotations) with message: "Usage of @Future annotation should be limited. Consider implementing the Queueable interface instead." This is a false-negative.

**Running PMD through:** CLI

**Comment by adangel** [comment]
Thanks for reporting. However, I can't reproduce this:

* freshly downloaded PMD from https://pmd.github.io/
* running the following script:

```shell
PMD_HOME="pmd-bin-7.20.0"

$PMD_HOME/bin/pmd --version
$PMD_HOME/bin/pmd check \
    --no-cache \
    --no-progress \
    -d src \
    -R category/apex/bestpractices.xml/AvoidFutureAnnotation \
    -f text

```

gives the following output:


```
  ████                            ████
  ██                                ██
  ██  █████ █ ███    ███  ███████   ██
 ███  ██  ███ ████  ████  ██    ██  ███
 ███  ███████ ██ ████ ██  ██    ██  ███
  ██  ██      ██  ██  ██  ███████   ██
  ██                                ██
  ████                            ████
PMD 7.20.0 (fa478ec141c154d34aac5389f0d5add0f28ecfea, 2025-12-30T14:37:52Z)
Java version: 25.0.1, vendor: Eclipse Adoptium, runtime: /home/andreas/programs/openjdk/eclipse-temurin/jdk-25.0.1+8
src/AtFutureRecipies.cls:3:     AvoidFutureAnnotation:  Usage of @Future annotation should be limited. Consider implementing the Queueable interface instead.
src/AtFutureRecipies.cls:8:     AvoidFutureAnnotation:  Usage of @Future annotation should be limited. Consider implementing the Queueable interface instead.
```

Where did you get your PMD installation? Can you give more information on how you actually call PMD? Which ruleset are you using? (Maybe there is some other error in the ruleset...)

**Comment by namrata111f** [comment]
Looks like an issue in older jar getting loaded on my end.

**Comment by namrata111f** [comment]
Thanks for taking a look. The ruleset did not have the latest rules due to multiple jar being loaded in my local.
